src/test/scala/org/broadinstitute/dsde/firecloud/EntityClientSpec.scala

```diff
@@ -0,0 +1,32 @@
+package org.broadinstitute.dsde.firecloud
+
+import java.util.zip.ZipFile
+
+import org.broadinstitute.dsde.firecloud.service.BaseServiceSpec
+import org.scalatest.BeforeAndAfterEach
+
+class EntityClientSpec extends BaseServiceSpec with BeforeAndAfterEach {
+  override def beforeEach(): Unit = {
+    searchDao.reset
+  }
+
+  override def afterEach(): Unit = {
+    searchDao.reset
+  }
+
+  "EntityClient should extract TSV files out of bagit zips" - {
```

Comment:
The only other thing I can think of is a test that participants are imported before samples. Not sure how to verify that, though... except maybe an integration test that has references that would fail if done in the wrong order.